app/models/user_checklist_manager.rb

```diff
@@ -4,113 +4,64 @@
-
-      guideline_list.children.select { |kid| !kid.completed? }&.sort_by { |kid| kid.list_position }&.first
-    else
-      nil
-    end
+    not_completed_guidelines_for(user)&.first
   end
 
 
-  # Is this user required to complete a Membership Guideline checklist?
-  # As of 2020-01-23,
-  # If someone is a current member as of < when the requirement starts >,
-  #   then they do _not_ have to complete the membership guidelines checklist until they renew.
-  #   else they DO have to complete the membership guidelines checklist.
-  #
-  # We only have payment dates and membership expiration dates to work with.
-  #
-  # Note that it is possible someone has paid ahead so that their membership term doesn't expire until
-  #   AFTER   1 membership term (= 1 year) after the requirement starts
-  #
-  #
-  def self.must_complete_membership_guidelines_checklist?(user)
-    raise ArgumentError, "User should not be nil. #{__method__}" if user.nil?
-
-    return false unless membership_guidelines_agreement_required_now?
-    return true unless user.membership_current?
-
-    # They are are current member AND the guidelines are required today
-
-    one_term_after_req_start_date = User.expire_date_for_start_date(membership_guidelines_reqd_start_date)
-
-    if user.membership_expire_date < one_term_after_req_start_date
-      false # today is before the membership expire date else they would no longer be a member.
-    else
-      # false if last payment made was _before_ the requirements start date
-      if user.most_recent_payment.created_at < membership_guidelines_reqd_start_date
-        false
-      else
-        true
-      end
-    end
-
+  def self.completed_guidelines_for(user)
+    guidelines_for(user, :completed)
   end
 
 
-  def self.membership_guidelines_reqd_start_date
-    # Could use a class variable here to store/cache ('memoize') this, but reading ENV and parsing the Time.zone is pretty fast.  Plus, this won't happen often.
-    if ENV.has_key?('SHF_MEMBERSHIP_GUIDELINES_CHECKLIST_REQD_START')
-      Time.zone.parse(ENV['SHF_MEMBERSHIP_GUIDELINES_CHECKLIST_REQD_START']).to_time
-    else
-      missing_membership_guidelines_reqd_start_date
-    end
+  def self.not_completed_guidelines_for(user)
+    # guideline_list.descendants.uncompleted.to_a
+    guidelines_for(user, :uncompleted)
   end
 
 
-  def self.missing_membership_guidelines_reqd_start_date
-    Time.zone.now.localtime.yesterday
-  end
+  # Make one if it doesn't exist?  NO.  Just return empty list.
+  # The selection_block is being used explictly so it's very clear.  (Could have also use yield.)
+  def self.guidelines_for(user, completed_method = :completed)
+    guideline_list = membership_guidelines_list_for(user) ? membership_guidelines_list_for(user) : {}
+    return [] unless guideline_list.present?
```

Comment:
I should know this, but how were "legacy" users given a checklist?  Was there a rake task that did this?

Reply:
They're created on demand.  Whenever a list is asked for, this method (essentially a `find_or_create`) is used. :
```
 def self.find_or_create_membership_guidelines_list_for(user)
    found_guidelines = membership_guidelines_list_for(user)
    found_guidelines.nil? ? AdminOnly::UserChecklistFactory.create_member_guidelines_checklist_for(user) : found_guidelines
  end
```